libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -295,12 +297,11 @@ func NewLeaderElectionManager(
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(ctx context.Context) {
 				logger.Debugf("leader election lock GAINED, id %v", id)
-				eventID := fmt.Sprintf("%v-%v", metaUID, time.Now().UnixNano())
+				eventID = fmt.Sprintf("%v-%v", leaseId, time.Now().UnixNano())
```

Comment:
can you log the eventID as well when the lock is gained and when it is lost?

Then it would be great if you could test this and then add the log lines with the same eventID to this PR